consensus/service/src/consensus_service.rs

```diff
@@ -603,21 +605,31 @@ impl<
                     block_height = Some(b);
                     latest_block_hash = ledger_db
                         .get_block(b - 1)
-                        .map(|x| format!("{:X?}", x.id.0))
+                        .map(|x| hex::encode(x.id.0))
                         .map_err(|e| log::error!(logger, "Error getting block {} {:?}", b - 1, e))
                         .ok();
-                    latest_block_timestamp = ledger_db
-                        .get_block_signature(b - 1)
-                        .map(|x| x.signed_at())
-                        .map_err(|e| {
+
+                    latest_block_timestamp = match ledger_db.get_block_signature(b - 1) {
+                        Ok(x) => Some(x.signed_at()),
+                        // Note, a block signature will be missing if the corresponding block was not
+                        // processed by an enclave participating in consensus. For example, unsigned
+                        // blocks can be created by a validator node that falls behind its peers and
+                        // enters into catchup. Another scenario would be if a validator node is not
+                        // trusted by sufficient peers to participate in quorum.
+                        Err(LedgerDbError::NotFound) => {
+                            log::debug!(logger, "Block signature not found for block {}", b - 1);
```

Comment:
I don't think this is necessary. Trace at most.

Reply:
👍